exp/exp.go

```diff
@@ -0,0 +1,10 @@
+package exp
+
+import (
+	"github.com/ethereum/go-ethereum/metrics"
+	e "github.com/ethereum/go-ethereum/metrics/exp"
+)
+
+func init() {
+	e.Exp(metrics.DefaultRegistry)
+}
```

Comment:
I don't think you wanted to commit this file :)

Reply:
Or I don't get it why this is outside in its own package vs underneath metrics?

Reply:
Currently we have https://github.com/ethereum/go-ethereum/blob/master/metrics/metrics.go#L48 . Since we are merging go-ethereum metrics and go-metrics in one package, we can't have this initialisation in the new joint `metrics` as it results in circular dependency.

Reply:
Then again I don't think we want a separate package just for this, so it can go anywhere else but in the `metrics` package.

Reply:
Well, since it's our custom package now, we can just do this call within `metrics/exp/exp.go` as an init. Would that be bad?

Reply:
Sure, I will move it to `metrics/exp/exp.go`. What I am not very happy about is that we are starting an HTTP server in this file, which is kind of hidden away, and we would always be initialising the /debug/metrics endpoint with the default registry.

I can imagine that at some point we might want to have more than one registry (default registry, swarm registry, etc.), but I guess for now every binary can just use the default registry for simplicity.

So for now `metrics/exp/exp.go` should be fine to init this server, but we might want to refactor in the future.

Reply:
This endpoint is exposed as part of the pprof http server (see https://github.com/ethereum/go-ethereum/commit/9e9bfc4e264b0bc16bdfe893e9cb3cfad096c48c), so I decided to set it up next to the pprof http server.

Moving it to `metrics/exp/exp.go` is not a solution, because no package is including `exp` so the `init` would never be called. We need another package to import the `exp` package, so `internal/debug` seems to be the correct place.

@karalabe I hope that makes sense.